src/components/AgreementContentFilter/AgreementContentFilter.js

```diff
@@ -0,0 +1,240 @@
+                          value: 'isNotEmpty',
+                          label: intl.formatMessage({
+                            id: 'ui-agreements.agreementContent.filter.has',
+                          }),
+                        },
+                        {
+                          value: 'isEmpty',
+                          label: intl.formatMessage({
+                            id: 'ui-agreements.agreementContent.filter.hasNot',
+                          }),
+                        },
+                      ]}
+                      name={`${filter}.attribute`}
+                      onChange={(e) => {
+                        change(`${filter}.attribute`, e?.target?.value);
+                        handleSubmit();
+                      }}
+                    />
+                    <Field
+                      component={MultiSelection}
+                      dataOptions={agreementContentOptions}
+                      name={`${filter}.content`}
+                      onChange={(e) => {
+                        change(`${filter}.content`, e);
+                        handleSubmit();
+                      }}
+                    />
+                  </Col>
+                  {index !== 0 && (
+                    <Col>
+                      <IconButton
+                        icon="times-circle-solid"
+                        onClick={() => {
+                          fields.remove(index);
+                          handleSubmit();
+                        }}
+                      />
+                    </Col>
+                  )}
+                </Row>
+              </>
+            );
+          })}
+          <Dropdown
+            label={
+              <FormattedMessage id="ui-agreements.agreementContent.filter.addFilter" />
+            }
+          >
+            <DropdownMenu>
+              <Button
+                buttonStyle="dropdownItem"
+                onClick={() => fields.push({ grouping: '&&' })}
+              >
+                <FormattedMessage id="ui-agreements.AND" />
+              </Button>
+              <Button
+                buttonStyle="dropdownItem"
+                onClick={() => fields.push({ grouping: '||' })}
+              >
+                <FormattedMessage id="ui-agreements.OR" />
+              </Button>
+            </DropdownMenu>
+          </Dropdown>
+        </>
+      )}
+    </FieldArray>
+  );
+};
+
+const AgreementContentFilter = ({
+  agreementContentFilters,
+  filterHandlers,
+  activeFilters,
+}) => {
+  const agreementContentOptions = useAgreementContentOptions();
+
+  // Used to map labels to content values for use within the multiselection
+  const mapContentLabels = (contentArray) => {
+    return contentArray.map((content) => {
+      return {
+        value: content,
+        label: agreementContentOptions?.find((e) => e?.value === content)
+          ?.label,
+      };
+    });
+  };
+
+  // Used to parse filters back from query string
```

Comment:
This is good, small comment with link to a JIRAS explaining whats needed from a genericised function pls then I'm happy with this